### PR TITLE
[FIX] #174 added timer operation, tests.

### PR DESCRIPTION
**Pull request by aricart**

### CODE REVIEW

File: lib/nats.js [hunk 1]
```diff
@@ -547,6 +595,8 @@ Client.prototype.createConnection = function() {
     }
     // Create the stream
     this.stream = net.createConnection(this.url.port, this.url.hostname);
+    this.stream.id = nuid.next();
+    this.emit('created', this.stream.id);
```
Comment (kozlovic) [inline lib/nats.js]: Do you need those 2 lines? I don't see the stream's 'created' event being used in the new test.

File: lib/nats.js [hunk 2]
```diff
@@ -999,7 +1052,8 @@ Client.prototype.processErr = function(s) {
     // except stale connection and permission errors
     var m = s ? s.toLowerCase() : '';
     if (m.indexOf(STALE_CONNECTION_ERR) !== -1) {
-        this.scheduleReconnect();
+        this.closeStream();
+        // this.scheduleReconnect();
```
Comment (kozlovic) [inline lib/nats.js]: Remove this and maybe instead add comment above to say that `closeStream()` will trigger reconnect if allowed.

File: test/support/mock_server.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    socket.write("PING\r\n");
+}
+
+exports.sendPing = sendPing;
+
+function sendPong(socket) {
+    socket.write("PONG\r\n");
+}
+
+exports.sendPong = sendPong;
```
Comment (kozlovic) [inline test/support/mock_server.js]: nit: do you need new line here?

Comment (aricart) [reply test/support/mock_server.js]: I added it as I showed you. For some reason, they are getting stripped out of the diff.

**Pull request merged**